tests/test_rpm_verifier.py

```diff
@@ -0,0 +1,54 @@
+"""Test rpm_verifier.py end-to-end"""
+
+from unittest.mock import MagicMock, call, create_autospec, sentinel
+
+import pytest
+from pytest import MonkeyPatch
+
+from verify_rpms import rpm_verifier
+from verify_rpms.rpm_verifier import ImageProcessor, generate_output
+
+
+class TestMain:
+    """Test call to rpm_verifier.py main function"""
+
+    @pytest.fixture()
+    def mock_image_processor(self, monkeypatch: MonkeyPatch) -> MagicMock:
+        """Monkey-patched ImageProcessor"""
+        mock: MagicMock = create_autospec(
+            ImageProcessor,
+            return_value=MagicMock(return_value=sentinel.output),
+        )
+        monkeypatch.setattr(rpm_verifier, ImageProcessor.__name__, mock)
+        return mock
+
+    @pytest.fixture()
+    def mock_generate_output(self, monkeypatch: MonkeyPatch) -> MagicMock:
+        """Monkey-patched db_getter"""
+        mock = create_autospec(generate_output)
+        monkeypatch.setattr(rpm_verifier, generate_output.__name__, mock)
+        return mock
+
+    def test_main(
```

Comment:
This kind of test will need to get rewritten every time we refactor main or one of its callees ,  lets try to structure the code and the tests in a way that avoids it.